File: fdaM/bspline/bspline_test.m
addpath('/Users/harry/Documents/MATLAB/fdaM')
addpath('/Users/harry/Documents/MATLAB/fdaM-Harry/bspline')

testbasis   = create_bspline_basis([0,2], 100, 4);
evalarg     = 0:0.01:2;

%tic;
basismat1   = getbasismatrix(evalarg, testbasis, 0)
%toc

%tic;
basismatC   = getbasismatrixC(evalarg, testbasis, 0)
%toc

subplot(2, 1, 1);
plot(basismat1);
subplot(2, 1, 2);
plot(basismatC);
max(max(abs(basismat1 - basismatC)))

File: fdaR/bspline/bspline_test.R
library(fda)
source("/Users/harry/Documents/MATLAB/fdaR-Harry/bspline/getbasismatrixC.R")

testbasis <- create.bspline.basis(c(0,1), 5)
evalarg <- seq(0, 1, 0.1)

basismat1 <- getbasismatrix(evalarg, testbasis, 0)
print(basismat1)

basismat2 <- getbasismatrixC(evalarg, testbasis, 0)
print(basismat1)

print(basismat1 - basismat2)